include/onnxruntime/core/framework/tensor.h

```diff
@@ -148,6 +148,12 @@ class Tensor final {
   */
   const OrtMemoryInfo& Location() const { return alloc_info_; }
 
+
+  /**
+    Reutrns the device type of the tensor's memory
+  */ 
+  int8_t DeviceType() const { return alloc_info_.device.Type();}
```

Comment:
i don't think we should add this api.
OrtMemoryInfo encapsulates the information about the memory including where it resides, device etc.
you can already get the MemoryInfo from a Tensor.

Reply:
Yes but I could not get the tensor from OrtValue , so either I can go through this route as I have implemented or add an API to get the Tensor from OrtValue. But I do feel there is a gap in API to get this information , Please suggest

Reply:
it's similar to what you've implemented, except you'd expose a capi to return OrtMemoryInfo of the Tensor. (see @ryanunderhill 's comment below)
then you wouldn't need to make the change here, since tensor.Location api is already available.

Reply:
I have made the changes , please review

Reply:
this change to tensor.h (addition of DeviceType() api) is not needed then, right?

Reply:
I have made the changes..